coq-sqir.opam

```diff
@@ -14,7 +14,7 @@ bug-reports: "https://github.com/inQWIRE/SQIR/issues"
 depends: [
   "dune" {>= "3.8"}
   "coq-interval" {>= "4.9.0"}
-  "coq-quantumlib" {= "1.3.0"}
+  "coq-quantumlib" {>= "1.5.0"}
```

Comment:
Do we want `1.5.1` here or `1.5.0` after the bump?

Reply:
`1.5.0` is correct. This PR includes an independent workaround for the bug fixed by `1.5.1`. It also includes a change to imports that enables / necessitates the bump to `1.5.0` (rather than `1.4.0`).